### PR TITLE
handle tzinfo in S3Hook.is_keys_unchanged_async

### PR DESCRIPTION
**Pull request by Lee-W**
When triggerer serialize datetime, it added tzinfo
https://github.com/apache/airflow/blob/1e6fa735752d61125903f0709b12cc1338789c5d/airflow/serialization/serializers/datetime.py#L51

which causes `can't compare offset-naive and offset-aware datetimes` in https://github.com/apache/airflow/blob/1e6fa735752d61125903f0709b12cc1338789c5d/airflow/providers/amazon/aws/hooks/s3.py#L716

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pankajkoti — APPROVED** [review]

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 1]
```diff
@@ -713,7 +713,9 @@ async def is_keys_unchanged_async(
             }
 
         if last_activity_time:
-            inactivity_seconds = int((datetime.now() - last_activity_time).total_seconds())
+            inactivity_seconds = int(
+                (datetime.now(last_activity_time.tzinfo) - last_activity_time).total_seconds()
```
Comment (dirrao) [inline airflow/providers/amazon/aws/hooks/s3.py:717]: I believe the last_activity_time initial value will come from the same function 1st call. There is no harm in considering the tzinfo from the other one. I am trying to understand here.

**Review by dirrao — COMMENTED** [review]
Thanks for the MR. I have one minor comment.

File: tests/providers/amazon/aws/hooks/test_s3.py [hunk 2]
```diff
@@ -812,6 +813,30 @@ async def test_s3_key_hook_is_keys_unchanged_inactivity_error_async(self, mock_l
...
+        response = await s3_hook_async.is_keys_unchanged_async(
+            client=mock_client.return_value,
+            bucket_name="test_bucket",
+            prefix="test",
+            inactivity_period=1,
+            min_objects=0,
+            previous_objects=set(),
+            inactivity_seconds=0,
+            allow_delete=False,
+            last_activity_time=std_datetime.now(timezone.utc),
```
Comment (hussein-awala) [inline tests/providers/amazon/aws/hooks/test_s3.py:862]: could you add a test with an offset-naive datetime?

**Review by hussein-awala — COMMENTED** [review]
It's ok to merge this PR for backward compatibility, but there is something to fix in the serialization. I will open a PR.

However, you need to ensure that it also works with offset-naive datetimes.

File: tests/providers/amazon/aws/hooks/test_s3.py (hunk 2 above)
Comment (Lee-W) [reply tests/providers/amazon/aws/hooks/test_s3.py:862]: Sure, just added

File: airflow/providers/amazon/aws/hooks/s3.py (hunk 1 above)
Comment (Lee-W) [reply airflow/providers/amazon/aws/hooks/s3.py:717]: Yep, they are supposed to be at the same timzone

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**